pkg/manifests/manifests.go

```diff
@@ -1214,6 +1218,16 @@ func (f *Factory) PrometheusK8s(host string, grpcTLS *v1.Secret, trustedCABundle
 		}
 	}
 
+	if f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs != nil {
+		p.Spec.AdditionalAlertManagerConfigs = &v1.SecretKeySelector{
+			Key: AdditionalAlertmanagerConfigKey,
+			LocalObjectReference: v1.LocalObjectReference{
+				Name: AdditionalAlertmanagerConfigName,
+			},
+		}
+		p.Spec.ConfigMaps = append(p.Spec.ConfigMaps, getConfigMapName(f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs)...)
```

Comment:
it's not clear to me why we need to mount an additional configmap?

Reply:
The additional configmap is used to store CA for the Alertmanager.

Reply:
Hmm so the idea is that a user has to 
1. Create a configmap `foo` in the `openshift-monitoring` namespace with key `ca.crt` that would hold the CA certificate.
2. Configure `CAFile: /etc/prometheus/configmaps/foo/ca.crt` in the cluster-monitoring-operator configmap?

Now the user has to know about where the file is mounted in the Prometheus pod. I'd rather see that the CA certificate content is stored directly in the CMO configmap (there's no security concern here) or `CAFile` is a configmap key selector in the CMO configmap. In both cases, CMO can generate the final configmap/secret, pass the generated configmap/secret to the Prometheus resource and inject the proper `ca_file` path in the additional Alertmanagers secret.

Reply:
Agreed that the user should not have to know about where the file is mounted in the Prometheus pod. How about have `CertsConfigmapName` in `TLSConfig`? if so, CMO can inject the proper `ca_file` `cert_file` `key_file` in the additional Alertmanagers secret. I think maybe MCO is not responsible for cert rotation. it should be owned by external process who generate the cert configmap. it is simple and easy to update the cert configmap by external process rather than `cluster-monitoring-config`. WDYT？